site/src/pages/WorkspacesPage/WorkspacesPage.tsx

```diff
@@ -22,20 +23,61 @@ const WorkspacesPage: FC = () => {
     ...pagination,
     ...filter,
   })
+
   const updateWorkspace = useWorkspaceUpdate(queryKey)
 
+  const [displayedWorkspaces, setDisplayedWorkspaces] = useState<Workspace[]>(
+    [],
+  )
+
+  useEffect(() => {
+    const fetchedWorkspaces = data?.workspaces || []
+    if (fetchedWorkspaces) {
+      if (displayedWorkspaces.length === 0) {
+        setDisplayedWorkspaces(fetchedWorkspaces)
+      } else {
```

Comment:
nit: I have a feeling that this can be refactored into more like a fluent approach.